centralstore/redis_store.go

```diff
@@ -567,10 +602,22 @@ func (r *RedisBasicStore) KeepTraces(ctx context.Context, statuses []*CentralTra
 
 // RecordTraceDecision records the decision made about a trace.
 // Note: Currently, the decision is only recorded in memory.
-func (r *RedisBasicStore) RecordTraceDecision(ctx context.Context, trace *CentralTraceStatus, keep bool, reason string) error {
+// The reason is decorated with a suffix to indicate that it is only stored in memory.
+func (r *RedisBasicStore) RecordTraceDecision(ctx context.Context, trace *CentralTraceStatus) error {
 	_, span := r.Tracer.Start(ctx, "RecordTraceDecision")
 	defer span.End()
 
+	var keep bool
+	switch trace.State {
+	case DecisionKeep:
+		keep = true
+	case DecisionDrop:
+		keep = false
+	default:
+		return fmt.Errorf("invalid trace state: %s", trace.State)
+	}
+	reason := trace.KeepReason + decisionKeepInMemoryOnly
```

Comment:
Do we have the ability to record a new flag on the status/trace? The temporary suffix feels too magical to me.

Reply:
Yes, I completely agree that it's a bit too magical. However, only redis_store cares about whether a decision is only stored in memory. I would prefer not have to change the cuckoo cache public interface because of this. We have plans to eventually move this cache into redis entirely. Once we do that, we won't need this magic anymore